pygmt/src/pygmtlogo.py

```diff
@@ -0,0 +1,266 @@
+            args_text_wm = {
+                "x": 0,
+                "y": -5,
+                "justify": "CT",
+                "font": f"2.5c,{font}",
+            }
+        case True | "horizontal":
+            args_text_wm = {
+                "x": 6,
+                "y": 0,
+                "justify": "LM",
+                "font": f"8c,{font}",
+            }
+
+    # -----------------------------------------------------------------------------
+    # Start plotting
+    # -----------------------------------------------------------------------------
+    fig = pygmt.Figure()
+
+    # .............................................................................
+    # blue circle / hexagon for Earth
+    # .............................................................................
+    fig.plot(
+        x=0,
+        y=0,
+        region=region,
+        projection=projection,
+        style=f"{symbol}{diameter}c",
+        pen=f"15p,{color_blue}",
+        fill=color_bg,
+        perspective=perspective,
+        no_clip=True,  # needed for corners of hexagon shape
+    )
+
+    # .............................................................................
+    # yellow lines for compass
+    # .............................................................................
+    lines_yellow = [
+        ([-size, size], [0, 0]),  # horizontal line
+        ([-r1, -r2], [r1, r2]),  # upper left
+        ([-r1, -r2], [-r1, -r2]),  # lower left
+        ([r1, r2], [r1, r2]),  # upper right
+        ([r1, r2], [-r1, -r2]),  # lower right
+    ]
+    for x, y in lines_yellow:
+        fig.plot(x=x, y=y, pen=f"5p,{color_yellow}", perspective=True)
+
+    # .............................................................................
+    # letter G
+    # .............................................................................
+    # horizontal red line
+    fig.plot(x=[0.1, 1.65], y=[0, 0], pen=f"12p,{color_red}", perspective=True)
+    # red ring sector
+    fig.plot(x=0, y=0, style="w3.3c/90/0+i2.35c", fill=color_red, perspective=True)
+    # space between yellow lines and ring sector
+    fig.plot(x=0, y=0, style="w3.7c/0/360+i3.3c", fill=color_bg, perspective=True)
+    # vertical yellow line
+    fig.plot(x=[0, 0], y=[-4, 4], pen=f"6p,{color_yellow}", perspective=True)
+    # cover yellow line in lower part of the ring sector
+    fig.plot(x=0, y=0, style="w3.3c/260/-80+i2.35c", fill=color_red, perspective=True)
+
+    # .............................................................................
+    # upper vertical red line
+    # .............................................................................
+    # space between red line and blue circle / hexagon
+    fig.plot(x=[0, 0], y=[4, 3.0], pen=f"18p,{color_bg}", perspective=True)
+    # red line
+    fig.plot(x=[0, 0], y=[4, 1.9], pen=f"12p,{color_red}", perspective=True)
+
+    # .............................................................................
```

Comment:
The letter M is also not perfect. 

<img width="707" alt="image" src="https://github.com/user-attachments/assets/4c72af08-408f-4934-b18a-36e28634ae0f" />

Instead of plotting four lines, maybe we could create a polygon?